text/0255-sensitive-data-redaction.md

```diff
@@ -0,0 +1,397 @@
+via `redaction`, where either a constant is selected that runs a predefined helper
+method, or a sed-like expression that uses a regular expression to apply redaction.
+
+Another possibility is to re-use a language like
+[OTTL](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/ottl/README.md).
+
+### Semantic Conventions
+
+To enable the redaction requirement profiles the semantic conventions need to be
+annotated with sensitivity details.
+
+Example:
+
+| Attribute | ... existing columns ... | sensitivity details |
+|-----------|--------------------------|---------------------|
+| `url.query`|                         | Rationale: Some verbatim wording why this is the way it is below<br>Type: `mixed`<br>`DEFAULT`: `REDACT_INSECURE_URL_PARAM_VALUES`<br>STRICT: `REDACT_ALL_URL_VALUES`<br>`STRICTER`: `DROP` |
+| `client.address`|                    | Rationale: some reasons why dropping octets may be required<br>Type: `maybe_pii`<br>`DEFAULT`: `NONE`<br>`STRICT`: `'s/([0-9]+\.[0-9]+\.[0-9]+\.)[0-9]+/\10/'`<br>`STRICTER`: `'s/([0-9]+\.[0-9]+\.)[0-9]+\.[0-9]+/\10.0/'` |
+| `enduser.creditCardNumber`**[1]** |     | Rationale: ...<br>Type: `always_pii`<br>DEFAULT: `EXTRACT_IIN`<br>`STRICT`: `DROP`|
+
+**[1]**: _This is a made-up example for demonstration purpose, it’s not part of the current semantic conventions. It gives a more nuanced example, e.g. that extracting the Issuer Identification Number (IIN) might be an option over dropping the number completely. It also demonstrated the value of “type”, which can enable Data lineage use cases_
+
+The `DROP` keyword means that the value is replaced with `REDACTED` (or set to null, …).
+
+It is the responsibility of the OpenTelemetry SDK to implement those sensitivity details provided by the semantic conventions.
+
+### Annotate attributes with sensitivity information
+
+As an additional building block the OpenTelemetry API should provide capabilities
+to library (and application) authors to apply in-place redaction
+
+One option is to add additional attributes to methods like `setAttribute`:
+
+```
+span.setAttribute("url.query", url.toString(), <SENSITIVITY_DETAILS>);
+```
+
+Or, if the method can not be extended, an additional method can be called:
+
+```
+span.setAttribute("url.query", OpenTelemetry.redact(url.toString(), <SENSITIVITY_DETAILS>);
+```
+
+Note that those API calls are no-op and will be implemented by the SDK (as we do it with other API methods as well), this way (almost?) no additional overhead will be created by introducing those annotations.
+
+The `<SENSITIVY_DETAILS>` should look like the following:
+
+```
+{
+  `<REDACTION_PROFILE>` => `<REDACTION_CONFIGURATION>`,
+  ...
+}
+```
+
+There may be multiple redaction profiles (like `DEFAULT`, `STRICT`, etc.), and
+for each one a configuration (as outlined above) may be applied.
+
+any sensitive data.
+
+## Internal details
+
+### Redactor
```

Comment:
Can you call out how this works across signals?

Right now we have no direct ties between Span/Metric/Trace via the SDK.  They actually only communicate with each other via `Context` API.